src/main/java/com/telefonica/iot/perseo/RulesServlet.java

```diff
@@ -151,8 +162,15 @@ protected void doDelete(HttpServletRequest request, HttpServletResponse response
         response.setContentType("application/json;charset=UTF-8");
         PrintWriter out = response.getWriter();
         logger.info("delete rule " + request.getPathInfo());
+        //request.getPathInfo() returns null or the extra path information
```

Comment:
What about if request.getPathInfo() retruns null? I undesratnad that `ruleName.substring(1)` will result in NullPointeException. Is that exception being managed?

Reply:
I have trying to add a new test following the model of those that already exist to verify this circumstance, but with the current behavior of the method of the Help class used for the requests, an empty path that returns a null can never arrive. So at least a "/" will arrive

But I think in the real world this case can happen. If I add a check for this null, the expected behavior would be that the server would return a 404 error, 405, other?
```suggestion
        if (ruleName == null) {
            response.setStatus(405);
            out.println("Method not alowed");
            out.close();
            return;
        }
```

Reply:
I don't remember the "small letter" :) of the HTTP RFC, but I'd say that 405 means that the URL is correct but the verb is wrong. In this case maybe a general 400 Bad Request is better.

Reply:
Yes, a 405 would be more appropriate when for example you try to do a PUT to an endpoint that does not accept this type of operation. I think a 400 is fine.